File: JAVA/Employee salary/com/Salary/Discounts.java
package com.Salary;


public class Discounts  {
    private double calculateARL;
    private double ARL;
    private Employee employee;
    private String[] levelRisk;
    private double pension;
    private double health;
    private double totalDiscounts;
    private double cDaytimeOvertime;
    private double cNightimeOvertime;
    private double cHolidaysOvertime;
    private double extras;
    private double hourlyRate;
    
    
   

    public Discounts(Employee employee, String[] levelRisk) {
        
        this.levelRisk=levelRisk;
        this.ARL = getCalculateARL(employee);
        this.employee = employee;
        this.pension = employee.getSalary() * 0.04;
        this.health = employee.getSalary() * 0.04;
        this.hourlyRate = employee.getSalary() / (employee.getDaysWorked()*8);
        this.cDaytimeOvertime = this.hourlyRate * 1.25 * employee.getDaytimeOvertime();
        this.cNightimeOvertime = this.hourlyRate * 1.75 * employee.getNightimeOvertime();
        this.cHolidaysOvertime = this.hourlyRate * 2 * employee.getHolidaysOvertime();
        this.totalDiscounts = this.ARL + this.pension + this.health;
        this.extras = this.cDaytimeOvertime + this.cNightimeOvertime + this.cHolidaysOvertime;
        
        
    }

    public double getCalculateARL(Employee employee){
        int numLevelRisk= this.levelRisk.length;

        for(int iteration=0; iteration < numLevelRisk; iteration++){
            String nlevelRisk=this.levelRisk[iteration];

           if(nlevelRisk.equalsIgnoreCase("minimo")){
               return employee.getSalary()* 0.00522;
           }

           else if(nlevelRisk.equalsIgnoreCase("bajo")){
                return employee.getSalary()* 0.01024;

           }

           else if(nlevelRisk.equalsIgnoreCase("medio")){
                return employee.getSalary()* 0.02436;

           }

           else if(nlevelRisk.equalsIgnoreCase("alto")){
                return employee.getSalary()*0.0435;
           }
        }
        return 0;
        
    }


    public double getARL(){
        return this.ARL;
    }

    public double getPension(){
        return this.pension;
    }

    public double getHealth(){
        return this.health;
    }
    public double getHourlyRate(){
        return this.hourlyRate;
    }


    public double getTotalDiscounts(){
        return this.totalDiscounts;
    }

    public double getCDaytimeOvertime() {
        return this.cDaytimeOvertime;
    }

    public double getCNightimeOvertime() {
        return this.cNightimeOvertime;
    }

    public double getCHolidaysOvertime() {
        return this.cHolidaysOvertime;
    }

    public double getExtras() {
        return this.extras;
    }

    public void show(){
        System.out.println("El pago por hora es de:" + this.getHourlyRate());
        System.out.println("El ARL equivale a: "+this.getARL());
        System.out.println("La pension equivale a: "+this.getPension());
        System.out.println("La salud equivale a: "+this.getHealth());
        System.out.println("El total de descuentos equivale a: "+this.getTotalDiscounts());
        System.out.println("El total de dinero generado en las horas extras diurnas equivale a: "+this.getCDaytimeOvertime());
        System.out.println("El total de dinero generado en las horas extras nocturnas equivale a: "+this.getCNightimeOvertime());
        System.out.println("El total de dinero generado en las horas extras festivas equivale a: "+this.getCHolidaysOvertime());
        System.out.println("El total de dinero generado en las horas extras equivale a: "+this.getExtras());
    }

}

File: PHP/Employee Salary/Salary/Discounts.php
<?php


include_once 'Employee.php';

class Discounts {
    private float $calculateARL;
    private float $ARL;
    private Employee $employee;
    private array $levelRisk;
    private float $pension;
    private float $health;
    private float $totalDiscounts;
    private float $cDaytimeOvertime;
    private float $cNightimeOvertime;
    private float $cHolidaysOvertime;
    private float $extras;
    private float $hourlyRate;

    public function __construct(Employee $employee, array $levelRisk) {
        $this->levelRisk = $levelRisk;
        $this->ARL = $this->getCalculateARL($employee);
        $this->employee = $employee;
        $this->pension = $employee->getSalary() * 0.04;
        $this->health = $employee->getSalary() * 0.04;
        $this->hourlyRate = $employee->getSalary() / ($employee->getDaysWorked() * 8);
        $this->cDaytimeOvertime = $this->hourlyRate * 1.25 * $employee->getDaytimeOvertime();
        $this->cNightimeOvertime = $this->hourlyRate * 1.75 * $employee->getNightimeOvertime();
        $this->cHolidaysOvertime = $this->hourlyRate * 2 * $employee->getHolidaysOvertime();
        $this->totalDiscounts = $this->ARL + $this->pension + $this->health;
        $this->extras = $this->cDaytimeOvertime + $this->cNightimeOvertime + $this->cHolidaysOvertime;
    }


    public function getCalculateARL($employee) {
        $numLevelRisk = count($this->levelRisk);

        for ($iteration = 0; $iteration < $numLevelRisk; $iteration++) {
            $nlevelRisk = $this->levelRisk[$iteration];

            if (strcasecmp($nlevelRisk, "minimo") == 0) {
                return $employee->getSalary() * 0.00522;
            } elseif (strcasecmp($nlevelRisk, "bajo") == 0) {
                return $employee->getSalary() * 0.01024;
            } elseif (strcasecmp($nlevelRisk, "medio") == 0) {
                return $employee->getSalary() * 0.02436;
            } elseif (strcasecmp($nlevelRisk, "alto") == 0) {
                return $employee->getSalary() * 0.0435;
            }
        }
        return 0;
    }
    
    public function getARL(): float {
        return $this->ARL;
    }

    public function getPension(): float {
        return $this->pension;
    }

    public function getHealth(): float {
        return $this->health;
    }

    public function getHourlyRate(): float{
         return $this->hourlyRate;
    }

    public function getTotalDiscounts(): float {
        return $this->totalDiscounts;
    }

    public function getCDaytimeOvertime(): float {
        return $this->cDaytimeOvertime;
    }

    public function getCNightimeOvertime(): float {
        return $this->cNightimeOvertime;
    }

    public function getCHolidaysOvertime(): float {
        return $this->cHolidaysOvertime;
    }

    public function getExtras(): float {
        return $this->extras;
    }

    public function show(): void {
        echo "El pago por horas es de:". $this->getHourlyRate() . "<br>";
        echo "El ARL equivale a: " . $this->getARL() . "<br>";
        echo "La pensión equivale a: " . $this->getPension() . "<br>";
        echo "La salud equivale a: " . $this->getHealth() . "<br>";
        echo "El total de descuentos equivale a: " . $this->getTotalDiscounts() . "<br>";
        echo "El total de dinero generado en las horas extras diurnas equivale a: " . $this->getCDaytimeOvertime() . "<br>";
        echo "El total de dinero generado en las horas extras nocturnas equivale a: " . $this->getCNightimeOvertime() . "<br>";
        echo "El total de dinero generado en las horas extras festivas equivale a: " . $this->getCHolidaysOvertime() . "<br>";
        echo "El total de dinero generado en las horas extras equivale a: " . $this->getExtras() . "<br>";
    }
}